core/util/src/main/java/org/eclipse/rdf4j/common/io/IOUtil.java

```diff
@@ -482,4 +483,49 @@ public static final long transfer(Reader reader, File file) throws IOException {
 			return transfer(reader, writer);
 		}
 	}
+
+	/**
+	 * Write an variable length integer.
+	 *
+	 * @param out   The {@link OutputStream} to write to.
+	 * @param value The {@code int} value to write.
+	 * @throws IOException If an error occurred while writing the integer.
+	 */
+	public static void writeVarInt(OutputStream out, int value) throws IOException {
+		if (value < 0) {
+			throw new IllegalArgumentException("Unable to write negative variable length integer");
+		}
+
+		while (value > 127) {
+			out.write(value & 0b01111111 | 0b10000000);
+			value >>>= 7;
+		}
+
+		out.write(value);
+	}
```

Comment:
I'm not completely following what your'e doing here, to be honest. Can you clarify how the varint format works in the javadoc?

Reply:
I've added an explanation in the [javadoc](https://github.com/eclipse/rdf4j/pull/3052/files#diff-13bedf2941dcdb65a0709af8c7f793ae867a61ef5da27721d97aec105bbcf69bR487). Let me know if this is satisfactory.

Reply:
Thanks, that's very helpful. Clever!

Reply:
Yes, always a pleasure to use tech from the '80 that still rocks :)